godot-core/src/builtin/meta/signature.rs

```diff
@@ -127,133 +107,104 @@ macro_rules! impl_signature_for_tuple {
 
             #[inline]
             unsafe fn varcall<C : GodotClass>(
-				instance_ptr: sys::GDExtensionClassInstancePtr,
-                args_ptr: *const sys::GDExtensionConstVariantPtr,
-                ret: sys::GDExtensionVariantPtr,
-                err: *mut sys::GDExtensionCallError,
-                func: fn(&C, Self::Params) -> Self::Ret,
-                method_name: &str,
-            ) {
-    	        $crate::out!("varcall: {}", method_name);
-
-                let storage = unsafe { crate::private::as_storage::<C>(instance_ptr) };
-                let instance = storage.get();
-
-                let args = ( $(
-                    {
-                        let variant = unsafe { &*(*args_ptr.offset($n) as *mut Variant) }; // TODO from_var_sys
-                        let arg = <$Pn as FromVariant>::try_from_variant(variant)
-                            .unwrap_or_else(|e| param_error::<$Pn>(method_name, $n, variant));
-
-                        arg
-                    },
-                )* );
-
-				let ret_val = func(&*instance, args);
-                let ret_variant = <$R as ToVariant>::to_variant(&ret_val); // TODO write_sys
-				unsafe {
-                    *(ret as *mut Variant) = ret_variant;
-                    (*err).error = sys::GDEXTENSION_CALL_OK;
-                }
-            }
-
-            #[inline]
-            unsafe fn varcall_mut<C : GodotClass>(
-				instance_ptr: sys::GDExtensionClassInstancePtr,
+                instance_ptr: sys::GDExtensionClassInstancePtr,
                 args_ptr: *const sys::GDExtensionConstVariantPtr,
                 ret: sys::GDExtensionVariantPtr,
                 err: *mut sys::GDExtensionCallError,
-                func: fn(&mut C, Self::Params) -> Self::Ret,
+                func: fn(sys::GDExtensionClassInstancePtr, Self::Params) -> Self::Ret,
                 method_name: &str,
             ) {
-    	        $crate::out!("varcall: {}", method_name);
+                $crate::out!("varcall: {}", method_name);
 
-                let storage = unsafe { crate::private::as_storage::<C>(instance_ptr) };
-                let mut instance = storage.get_mut();
+                let args_ptr = args_ptr as *const *mut Variant;
```

Comment:
This could probably be moved  inside `varcall_arg` -- the conversion happens at compile time, so it doesn't matter how many elements need  to do it.

That way, the access logic is more local, and to the outside we have only an opaque `sys` pointer.

Reply:
`varcall_arg` isn't public so it wont be visible from the outside. and this way it's clear from the signature that we are intending for this to be a pointer to an actual `Variant` that we are going to dereference. but i can still do it if you want.